docs/using-the-compiler.rst

```diff
@@ -201,14 +201,23 @@ Input Description
           }
         }
         // The following can be used to select desired outputs.
-        // If this field is omitted, then the compiler loads and does type checking, but will not generate any outputs apart from errors.
-        // The first level key is the file name and the second is the contract name, where empty contract name refers to the file itself,
+        // If this field is omitted, then the compiler loads and does type checking,
+        // but will not generate any outputs apart from errors.
+        // The first level key is the file name and the second is the contract name,
+        // where empty contract name refers to the file itself,
```

Comment:
**an** empty contract name?
And actually: what is "refers to the file itself" mean?
I would split the whole sentence:
```
The first level key is the file name and the second level key is the contract name.
An empty contract name refers to the contract with the same name as the file.
A star as contract name refers to all contracts in the file.
```
I guess that is what is meant?

Reply:
No, it's not :)
The thing is that outputs like the AST cannot be tied to a contract, they have to be tied to a file. On the other hand, a contract's bytecode cannot be tied to a file, it can only be tied to a contract (which might even span multiple files).

Reply:
Ah ok :-). Yes that actually makes more sense.
Then I would make that explicit, though, e.g. as a replacement of the second sentence in my suggestion above:
``An empty contract name is used for output selections that are not tied to a contract but to the whole source file.`` (or something similar)